internal/transport/http2_server.go

```diff
@@ -1062,23 +1075,13 @@ func (t *http2Server) WriteStatus(s *Stream, st *status.Status) error {
 func (t *http2Server) Write(s *Stream, hdr []byte, data []byte, opts *Options) error {
 	if !s.isHeaderSent() { // Headers haven't been written yet.
 		if err := t.WriteHeader(s, nil); err != nil {
-			if _, ok := err.(ConnectionError); ok {
-				return err
-			}
-			// TODO(mmukhi, dfawley): Make sure this is the right code to return.
-			return status.Errorf(codes.Internal, "transport: %v", err)
+			return err
 		}
 	} else {
 		// Writing headers checks for this condition.
 		if s.getState() == streamDone {
 			// TODO(mmukhi, dfawley): Should the server write also return io.EOF?
```

Comment:
I think this can be deleted now.

Reply:
The TODO or the entire conditional? The tests appear to pass with the entire conditional removed but I'm not familiar enough with this code to understand whether we still need to check if the stream is done elsewhere in `Write` (if it's called after writing headers).

Reply:
It looks like `writeQueue.get` checks whether the `writeQueue` is `done`. Is the idea that that channel is set when the stream is closed?

Reply:
I was just referring to the TODO. I don't think this whole thing should be removed or else we'll potentially write operations into the control buffer for closed streams.  If the controlbuf isn't checking that the stream is still valid and writes frames, then that's an HTTP/2 protocol violation.

Reply:
ack